src/imports.test.ts

```diff
@@ -0,0 +1,219 @@
+				name: "testEnum",
+				canonical_name: "test",
+				dimensions: 1,
+				enum_values: ["test1", "test2", "test3"],
+				original_type: "string",
+			},
+		});
+
+		expect(importInstance).toBeInstanceOf(Import);
+		expect(
+			typeof importInstance.from === "function" ||
+				typeof importInstance.from === "string",
+		).toBe(true);
+
+		if (typeof importInstance.from === "function") {
+			const generatedPath = importInstance.from(files);
+
+			expect(generatedPath).toBe("../enums/testEnum.ts");
+		}
+	});
+
+	it("should handle missing `namedImports` or other constructor properties", () => {
+		const importInstance = new Import({ from: "module/path" });
+
+		expect(importInstance.from).toBe("module/path");
+		expect(importInstance.namedImports).toBeUndefined();
+		expect(importInstance.star).toBeUndefined();
+		expect(importInstance.default).toBeUndefined();
+		expect(importInstance.typeOnly).toBe(false);
+	});
+});
+
+describe("ImportList Class", () => {
+	it("should add imports correctly", () => {
+		const importList = new ImportList();
+		const newImport = new Import({
+			from: "module/path",
+			namedImports: ["MyComponent"],
+		});
+
+		importList.add(newImport);
+		expect(importList.imports).toHaveLength(1);
+		expect(importList.imports[0]).toBe(newImport);
+	});
+
+	it("should merge import lists correctly", () => {
+		const list1 = new ImportList([
+			new Import({ from: "module1", namedImports: ["a"] }),
+		]);
+		const list2 = new ImportList([
+			new Import({ from: "module2", namedImports: ["b"] }),
+		]);
+
+		const mergedList = ImportList.merge([list1, list2]);
+		expect(mergedList.imports).toHaveLength(2);
+	});
+
+	it("should merge import lists correctly with empty list", () => {
+		const list1 = new ImportList([
+			new Import({ from: "module1", namedImports: ["a"] }),
+		]);
+		const list2 = new ImportList([]);
+
+		const mergedList = ImportList.merge([list1, list2]);
+		expect(mergedList.imports).toHaveLength(1);
+	});
+
+	it("should stringify imports correctly", () => {
+		const importList = new ImportList([
+			new Import({ from: "module1", namedImports: ["a"] }),
+			new Import({ from: "module1", namedImports: ["b"] }),
+		]);
+
+		const result = importList.stringify(files);
+
+		expect(result).toContain('import { a, b } from "module1";');
+	});
+
+	it("should handle empty ImportList gracefully", () => {
+		const importList = new ImportList();
+		const files: FolderStructure = {
+			name: "root",
+			type: "root",
+			children: {},
+		};
+
+		const result = importList.stringify(files);
+		expect(result).toBe("");
+	});
+
+	it("should handle duplicate imports and avoid repetition", () => {
+		const importList = new ImportList([
+			new Import({ from: "module1", namedImports: ["a"] }),
+			new Import({ from: "module1", namedImports: ["a"] }),
+		]);
+
+		const result = importList.stringify(files);
+		expect(result).toContain('import { a } from "module1";');
+		expect(result.split("import").length).toBe(2);

```

Comment:
This last test is a bit weird. We probably just want `expect(result).toBe(...)` in the previous line since only one line will be generated? And maybe the last test would be `result.split("\n").length` to be 1 to check that there are only 1 lines?